src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue

```diff
@@ -3,20 +3,25 @@
          class="dashboard-widget-container"
     >
         <div
-            v-for="(row, rowIndex) in widgetWidthList"
+            v-for="(rowWidgetList, rowIndex) in containerWidgetConfigList"
             :key="`widget-row-${rowIndex}`"
             class="dashboard-widget-row"
         >
-            <a-w-s-cloud-front-cost
-                v-for="(width, colIndex) in row"
-                ref="widgetRef"
-                :key="`widget-row-${rowIndex}-${colIndex}`"
-                :widget-key="`widget-row-${rowIndex}-${colIndex}`"
-                widget-config-id="awsCloudFrontCost"
-                :width="width"
-                :is-full="containerWidth === width"
-                :widget-index="widgetIndexList[rowIndex][colIndex]"
-                @click-expand-icon="handleExpand"
+            <component :is="widgetConfig.widget_component"
+                       v-for="(widgetConfig, colIndex) in rowWidgetList"
+                       :key="`widget-row-${rowIndex}-${colIndex}`"
+                       ref="widgetRef"
+                       :widget-config-id="widgetConfig.widget_config_id"
+                       :widget-key="widgetConfig.widget_config_id"
+                       :title="widgetConfig.title"
+                       :options="widgetConfig.widget_options"
+                       :inherit-options="{
+                           currency: { enabled: true },
+                           date_range: { enabled: true },
+                       }"
+                       :size="widgetConfig.size"
+                       :width="widgetConfig.width"
+                       :theme="widgetConfig.themeName"
             />
         </div>
     </div>
```

Comment:
suggestion: I strongly recommend to use just one dimensioned array for drag&drop feature.